tests/unit/net/fetch.rs

```diff
@@ -506,6 +512,55 @@ fn test_fetch_with_local_urls_only() {
     assert!(server_response.is_network_error());
 }
 
+#[test]
+fn test_fetch_with_hsts() {
+    static MESSAGE: &'static [u8] = b"";
+    let handler = move |_: HyperRequest, response: HyperResponse| {
+        response.send(MESSAGE).unwrap();
```

Comment:
Let's have the response contain the scheme of the request's URL instead. This will let us verify that the server never receives a non-HTTPS connection.